source/type/pb_converter.cpp

```diff
@@ -1017,7 +1017,9 @@ void PbCreator::Filler::fill_message(const boost::shared_ptr<nd::Impact>& impact
     if (none_of(pb_object->impact_uris(), [&](const std::string& uri) { return impact->uri == uri; })) {
         *pb_object->add_impact_uris() = impact->uri;
     }
-    pb_creator.impacts.insert(impact);
+    if (!pb_creator.disable_disruption) {
```

Comment:
It's better to shortcircuit at the start of the function.
In line 1018 we add the uri of the impact to the object, with this change we do not put the disruption be still have the link to it.